src/GmailServiceProvider.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Ushahidi\Gmail;
+
+use Illuminate\Support\ServiceProvider;
+
+class GmailServiceProvider extends ServiceProvider
+{
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        // Main Service
+        $this->app->bind('gmail', function ($app, $givenConfig = null) {
+            $config = $givenConfig ?: $app['config']['services.gmail'];
+            $gmail = new Gmail($config);
+            $gmail->setStorage($this->app->make(TokenDiskStorage::class));
+            return $gmail;
+        });
+
+        $this->app->bind(Gmail::class, function ($app) {
+            return $app->make('gmail');
+        });
+
+        $this->registerGmailSource();
+
+        $this->registerGmailTransport();
+
+        $this->registerCommands();
+    }
+
+    /**
+     * Register the Gmail data source driver.
+     *
+     * @return void
+     */
+    private function registerGmailSource()
+    {
+        if (! $this->shouldRegisterGmailSourceDriver()) {
+            return;
+        }
+
+        $this->app['datasources']->extend('gmail', function ($config) {
+            $config['user'] = 'wisdomaebong@gmail.com';
```

Comment:
This is just for test purpose, the $config['user'] will be provided by the config repository.